File: 3_roman_int/Main.java
import java.util.HashMap;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> romanToIntMap = new HashMap<>();
        romanToIntMap.put('I', 1);
        romanToIntMap.put('V', 5);
        romanToIntMap.put('X', 10);
        romanToIntMap.put('L', 50);
        romanToIntMap.put('C', 100);
        romanToIntMap.put('D', 500);
        romanToIntMap.put('M', 1000);
        
        int currentValue = 0;
        int previousValue = 0;
        int total = 0;
        
        for (int i = s.length() - 1; i >= 0; i--) {
            char currentChar = s.charAt(i);
            currentValue = romanToIntMap.get(currentChar);
            if (currentValue < previousValue) {
                total -= currentValue;
            } else {
                total += currentValue;
            }
            previousValue = currentValue;
        }
        return total;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        String romanNumeral = "MCMXCIV"; // Example input
        int result = solution.romanToInt(romanNumeral);
        System.out.println("The integer value of " + romanNumeral + " is: " + result);
    }
}

File: 3_roman_int/main.py
class Solution(object):
    def romanToInt(self, s):
        roman_to_int = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        curr = 0
        prev = 0
        total = 0
        for char in s[::-1]:
            curr = roman_to_int[char]
            if curr < prev:
                total -= curr
            else:
                total += curr
            prev = curr
        return total

if __name__ == "__main__":
    s = Solution()
    print(s.romanToInt("III"))  # Output: 3
    print(s.romanToInt("IV"))   # Output: 4
    print(s.romanToInt("IX"))   # Output: 9
    print(s.romanToInt("LVIII"))  # Output: 58
    print(s.romanToInt("MCMXCIV"))  # Output: 1994